apps/api/src/app/workflows-v2/mappers/workflow-template-get-mapper.ts

```diff
@@ -0,0 +1,77 @@
+import { NotificationStepEntity, NotificationTemplateEntity } from '@novu/dal';
+
+import { MinifiedResponseWorkflowDto, StepResponseDto, WorkflowResponseDto } from '../dto/workflow.dto';
+
+export class WorkflowTemplateGetMapper {
+  static toResponseWorkflowDto(template: NotificationTemplateEntity): WorkflowResponseDto {
+    return {
+      _id: template._id,
+      tags: template.tags,
+      active: template.active,
+      critical: template.critical,
+      notificationGroupId: template._notificationGroupId,
+      preferences: WorkflowTemplateGetMapper.toPreferences(template.preferenceSettings), // Assuming this directly maps; may need adjustment
+      steps: template.steps.map(WorkflowTemplateGetMapper.toStepResponseDto),
+      name: template.name,
+      description: template.description,
+      origin: template.origin || WorkflowOriginEnum.NOVU,
+      updatedAt: template.updatedAt || 'Missing Updated At',
+    };
+  }
+  static toMinifiedWorkflowDto(template: NotificationTemplateEntity): MinifiedResponseWorkflowDto {
+    return {
+      _id: template._id,
+      name: template.name,
+      tags: template.tags,
+      updatedAt: template.updatedAt || 'Missing Updated At',
+      stepTypeOverviews: template.steps.map(WorkflowTemplateGetMapper.buildStepSummery),
+    };
+  }
+
+  static toWorkflowsMinifiedDtos(templates: NotificationTemplateEntity[]): MinifiedResponseWorkflowDto[] {
+    return templates.map(WorkflowTemplateGetMapper.toMinifiedWorkflowDto);
+  }
+  static toStepResponseDto(step: NotificationStepEntity): StepResponseDto {
+    return {
+      name: step.name || 'Missing Name',
+      stepUuid: step._templateId,
+      type: step.template?.type || StepTypeEnum.EMAIL,
+      controls: WorkflowTemplateGetMapper.convertControls(step),
+      active: step.active || true,
+      shouldStopOnFail: step.shouldStopOnFail || true,
+      controlValues: step.controlVariables || {},
+    };
+  }
+
+  private static convertControls(step: NotificationStepEntity) {
+    if (step.template?.controls) {
+      return { schema: step.template.controls.schema };
+    } else {
+      return undefined;
+    }
+  }
+
+  private static toPreferences(preferenceSettings: IPreferenceChannels): DiscoverWorkflowOutputPreferences {
+    const { email } = preferenceSettings;
+    const { sms } = preferenceSettings;
+    const { push } = preferenceSettings;
+    const { chat } = preferenceSettings;
+    const inApp = preferenceSettings.in_app;
+    return {
+      workflow: { defaultValue: true, readOnly: false },
+      channels: {
+        email: { defaultValue: email ?? true, readOnly: false },
+        sms: { defaultValue: sms ?? true, readOnly: false },
+        push: { defaultValue: push ?? true, readOnly: false },
+        chat: { defaultValue: chat ?? true, readOnly: false },
+        in_app: { defaultValue: inApp ?? true, readOnly: false },
+      },
+    };
+  }
+
+  private static buildStepSummery(step: NotificationStepEntity) {
```

Comment:
A few issues:
1. `Summery` is mispelt here, it should be `Summary`
2. There is no summary returned, it's just a `string` right now. Please rename the method appropriately or return a summary of data.
3. Why is email used as the fallback step type? Assuming email could present many issues to the client when understanding the response, it's probably better to just throw an error if there is no `type` present
4. Please add a return type to the method.

Reply:
will choose a better name, maybe overview, the reason for the defualts is a decision in the Entity to have the type as optional although it's not, was trying to avoid touching entity as a way to limit scope, I'll look for a way to make it more clean